lib/gtfs/block.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Gtfs.Block do
+  alias Gtfs.Helpers
+  alias Gtfs.Trip
+
+  @type id :: String.t()
+  @type t :: [Trip.t()]
+  @type by_id :: %{{id(), Service.id()} => t()}
+
+  @spec group_trips_by_block([Trip.t()]) :: by_id()
+  def group_trips_by_block(trips) do
```

Comment:
Another variation to consider (sorry for the rambling as I think this over):

    @type block_id:: String.t()
    @type service_id :: String.t()
    @type id :: {block_id(), service_id()}
    @type by_id :: %{id() => t()}

(The Service module could go away as well)

Reply:
I guess the confusing part is that blocks aren't uniquely determined by their block_id. They're determined by the `block_id, service_id` pair. So which should be called `id`? I think, since `block_id` is an existing concept, we should keep it, and avoid making `Block.id()` refer to a different concept.

Pulling the map type and the get function into `Block` was my attempt to isolate the outside world from this confusion, but it doesn't remove the confusion from inside the `Block` module.

What do you think about just adding a `@doc` explaining why you need a service id to get a block?

Reply:
I think that the unique determiner for a module should be the ID, which makes me think that `@type id :: {block_id(), service_id()}` is the important part. What if we named this module something else other than block, so we have `block_id` and `service_id` but those don't need to reference other modules?

Reply:
`key` is a better name for the `{block_id(), service_id()}` tuple we just came up with.